python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -158,14 +158,27 @@ def dist_minibatch_inference(g, gnn_encoder, get_input_embeds, batch_size, fanou
                                                             shuffle=False,
                                                             drop_last=False)
 
-        for iter_l, (input_nodes, output_nodes, blocks) in enumerate(dataloader):
+        len_dataloader = max_num_batch = len(list(dataloader))
+        tensor = th.tensor([len_dataloader], device=device)
+        if is_distributed():
+            th.distributed.all_reduce(tensor, op=th.distributed.ReduceOp.MAX)
+            max_num_batch = tensor[0]
+        dataloader_iter = iter(dataloader)
+
+        # WholeGraph does not support imbalanced batch numbers across processes/trainers
+        # TODO (IN): Fix dataloader to have same number of minibatches.
+        for iter_l in range(max_num_batch):
+            if iter_l < len_dataloader:
+                input_nodes, output_nodes, blocks = next(dataloader_iter)
+            else:
+                for ntype in g.ntypes:
+                    input_nodes[ntype] = output_nodes[ntype] = th.empty((0,), dtype=g.idtype)
```

Comment:
```suggestion
                    input_nodes[ntype] = th.empty((0,), dtype=g.idtype)
```
We don't need to set output_nodes[ntype]. As in L195, we will skip the current interation.

Reply:
Fixed.